packages/devtools_app/lib/src/charts/treemap.dart

```diff
@@ -1,70 +1,56 @@
-import 'package:devtools_app/src/ui/colors.dart';
 import 'package:flutter/material.dart';
 
 import '../trees.dart';
+import '../ui/colors.dart';
 import '../utils.dart';
 
 enum PivotType { pivotByMiddle, pivotBySize }
 
-class Treemap extends StatefulWidget {
-  const Treemap({
-    @required this.rootNode,
-    @required this.levelsVisible,
-    @required this.width,
-    @required this.height,
-    @required this.onTap,
-  });
+class NewTreemap extends StatefulWidget {
+  NewTreemap._({
+    this.rootNode,
+    this.nodes = const [],
+    this.levelsVisible,
+    this.onRootChangedCallback,
+  }) : assert(rootNode == null && nodes.isNotEmpty ||
+            rootNode != null && nodes.isEmpty);
+
+  NewTreemap.fromRoot({
+    @required TreemapNode rootNode,
+    @required levelsVisible,
+    @required onRootChangedCallback,
+  }) : this._(
+            rootNode: rootNode,
+            levelsVisible: levelsVisible,
+            onRootChangedCallback: onRootChangedCallback);
+
+  NewTreemap.fromNodes({
+    @required List<TreemapNode> nodes,
+    @required levelsVisible,
+    @required onRootChangedCallback,
+  }) : this._(
+            nodes: nodes,
+            levelsVisible: levelsVisible,
+            onRootChangedCallback: onRootChangedCallback);
 
   final TreemapNode rootNode;
 
-  /// The depth of children visible from this Treemap widget.
-  ///
-  /// For example, levelsVisible = 2 at root Treemap:
-  /// ```
-  /// _______________
-  /// |     Root    |
-  /// ---------------
-  /// |    l = 1    |
-  /// |  ---------  |
-  /// |  | l = 2 |  |
-  /// |  |       |  |
-  /// |  |       |  |
-  /// |  ---------  |
-  /// ---------------
-  /// ```
-  final int levelsVisible;
-
-  final double width;
+  final List<TreemapNode> nodes;
 
-  final double height;
+  final int levelsVisible;
 
-  final VoidCallback onTap;
+  final void Function(TreemapNode) onRootChangedCallback;
```

Comment:
This should be 
`Function(TreemapNode node)`
Unless it got cleaned up when the `Function` keyword was added,
`Function(TreemapNode)`
actually means
`Function(dynamic TreemapNode)`